### DISCUSSION TITLE
Flux space sampling for Human1

### DISCUSSION THREAD

**Discussion by Bastien-N** [SysBioChalmers/Human-GEM#433] [Q&A]
Hi everyone,

I have been trying to perform flux space sampling on the Human1 model for the past week and I cannot seem to make it work. Normally I would just use CHRR from the Cobra-toolbox, but this does not seem to work, I suspect the model is too large for the rounding step to work. 

Other samplers are generally exponentially more expensive in time and computation, so they would be very impractical for my project.
Has anyone been able to perform sampling on Human1, and if so could you share which tool you used to do so?

Bastien

**Comment by haowang-bioinfo** [comment]
@Bastien-N good to know that you are using our model.

What are the error messages when running CHRR, or it took too long to finish?

**Reply by Bastien-N** [reply]
Thank you for the swift reply!
The error is:
"
Error using mve_solver_cobra (line 29)
x0 not interior

Error in mve_run_cobra (line 32)
[x,E2,converged] = mve_solver_cobra(A,b,x0,maxiter,tol2,reg); %Ben

Error in preprocess (line 255)
[T_shift, Tmve,converged] = mve_run_cobra(P.A,P.b, x0,reg);

Error in chrrSampler(line 110)
[roundedPolytope] = preprocess(P,options);
"
I have investigated and it appears that while trying to round the polytope for the model, CHRR cannot find any point inside the polytope after scaling. 
I also made an issue for this on the CobraToolbox repository, but figured that maybe someone here had experience trying to sample Human1.

**Comment by JonathanRob** [comment]
@Bastien-N, if it is potentially due to the large size of Human1, you could try again using a subset of the model (e.g., remove a few thousand reactions). Though you wouldn't want the model to become too disconnected, so the choice of which reactions to remove would be important.

**Reply by Bastien-N** [reply]
The aim of my project is to extract a cardiac-specific model from Human1, so if this is the solution I suppose it might fix itself. Unfortunately I would only know this after extraction, which is a long process.

**Reply by JonathanRob** [reply]
Which part of the extraction process are you expecting to be long? If you're planning to use tINIT, @johan-gson recently developed and released ftINIT (see [user guide](https://sysbiochalmers.github.io/Human-GEM-guide/gem_extraction/)), which is a **much** faster (and more accurate) algorithm than the previous version.

**Reply by Bastien-N** [reply]
I was intending to use mCADRE, as recommended by a more experienced colleague. For Human1 the runtime can apparently extend to several days from what I have seen in his project. I was, however, not aware of ftINIT. I know my colleague had some accuracy (gaps) issues with tINIT but if this was improved this could be a good alternative.

**Reply by JonathanRob** [reply]
Ok, mCADRE is also a good choice, though I understand that it may be slower for such a large model. If for some reason you end up trying both methods and comparing, we would of course be happy to hear about your findings 😬

**Comment by johan-gson** [comment]
Hi Bastien,

For ftINIT, there is a preparation step that takes roughly an hour, which
you only need to run once, i.e., not per sample. Then ftINIT usually takes
1-2 minutes per sample, depending on your computer. In the ftINIT paper
(under revision), we have for example generated 22,000 models, which didn't
take that long. One thing to think about if you want to try it is that
there are two ways to run it, in 1+0 mode and 1+1 mode. In 1+0, most
reactions without GPRs are kept in the model, while only the ones "needed"
are kept in the 1+1 mode. So, it depends a bit on what you want to do with
your model. For structural comparisons 1+0 is better (less randomness),
while 1+1 may be better in your case. But try both, it only takes a minute
extra to run.
